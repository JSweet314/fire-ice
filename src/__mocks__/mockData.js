/* eslint-disable max-len*/
export const mockHouses = [{
  ancestralWeapons: ["Lady Forlorn"],
  cadetBranches: [],
  coatOfArms: "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
  currentLord: "https://www.anapioficeandfire.com/api/characters/681",
  diedOut: "",
  founded: "",
  founder: "https://www.anapioficeandfire.com/api/characters/256",
  heir: "https://www.anapioficeandfire.com/api/characters/677",
  name: "House Corbray of Heart's Home",
  overlord: "https://www.anapioficeandfire.com/api/houses/7",
  region: "",
  seats: ["Heart's Home"],
  swornMembers: [
    "https://www.anapioficeandfire.com/api/characters/255", "https://www.anapioficeandfire.com/api/characters/256", "https://www.anapioficeandfire.com/api/characters/447", "https://www.anapioficeandfire.com/api/characters/654", "https://www.anapioficeandfire.com/api/characters/677", "https://www.anapioficeandfire.com/api/characters/681", "https://www.anapioficeandfire.com/api/characters/871", "https://www.anapioficeandfire.com/api/characters/1182"
  ],
  titles: ["Lord of Heart's Home", "King of the Fingers (historical)"],
  url: "https://www.anapioficeandfire.com/api/houses/81",
  words: ""
}];

export const mockCleanHouses = [{ "ancestralWeapons": "Lady Forlorn", "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)", "founded": "", "name": "House Corbray of Heart's Home", "seats": "Heart's Home", "swornMembers": ["255", "256", "447", "654", "677", "681", "871", "1182"], "titles": ["Lord of Heart's Home", "King of the Fingers (historical)"], "words": "" }];

export const mockCharacter = {
  "url": "https://www.anapioficeandfire.com/api/characters/255",
  "name": "Corwyn Corbray",
  "gender": "Male",
  "culture": "",
  "born": "",
  "died": "In 134 AC, at Runestone",
  "titles": [
    "Ser",
    "Lord Regent"
  ],
  "aliases": [
    ""
  ],
  "father": "",
  "mother": "",
  "spouse": "https://www.anapioficeandfire.com/api/characters/871",
  "allegiances": [
    "https://www.anapioficeandfire.com/api/houses/81"
  ],
  "books": [
    "https://www.anapioficeandfire.com/api/books/11"
  ],
  "povBooks": [],
  "tvSeries": [
    ""
  ],
  "playedBy": [
    ""
  ]
};

export const mockCleanCharacter = { "died": "In 134 AC, at Runestone", "name": "Corwyn Corbray" };