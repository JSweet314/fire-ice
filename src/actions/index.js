export const getHouses = () => ({
  type: 'GET_HOUSES'
});

export const getHousesSuccess = houses => ({
  type: 'GET_HOUSES_SUCCESS',
  houses
});

export const captureErrorMessage = errorMessage => ({
  type: 'CAPTURE_ERROR_MESSAGE',
  errorMessage
});
