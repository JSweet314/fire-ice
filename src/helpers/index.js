export const housesDataWrangler = housesArray => 
  housesArray.map(house => ({
    name: house.name,
    founded: house.founded,
    seats: house.seats.join(', '),
    titles: house.titles,
    coatOfArms: house.coatOfArms,
    ancestralWeapons: house.ancestralWeapons.join(', '),
    words: house.words
  }));