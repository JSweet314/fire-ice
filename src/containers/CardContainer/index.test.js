import React from 'react';
import { shallow } from 'enzyme';
import {CardContainer, mapDispatchToProps, mapStateToProps} from './index';
import * as actions from '../../actions';
import * as mockData from '../../__mocks__/mockData';
import * as helpers from '../../helpers';

describe('CardContainer', () => {
  let wrapper;
  const mockGetHouses = jest.fn();
  const mockGetMembers = jest.fn();
  const mockRemoveMembersDisplay = jest.fn();
  const mockAddMembersDisplay = jest.fn();
  const mockCardGenerator = jest.fn();
  const mockHouses = helpers.housesDataWrangler(mockData.mockHouses);
  beforeEach(() => {
    wrapper = shallow(
      <CardContainer 
        getHouses={mockGetHouses}
        getMembers={mockGetMembers}
        houses={mockHouses}
        memberDisplay={{white: true}}
        removeMembersDisplay={mockRemoveMembersDisplay}
        addMembersDisplay={mockAddMembersDisplay}/>
    );
  });

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should get houses after component mounts', () => {
    expect(mockGetHouses).toHaveBeenCalled();
  });

  it('should be able to remove a house from members display', () => {
    wrapper.instance().handleOnClick({name: 'white', members: []});
    expect(mockRemoveMembersDisplay).toHaveBeenCalled();
  });

  it('should be able to add a house to members display', () => {
    wrapper = shallow(
      <CardContainer
        getHouses={mockGetHouses}
        getMembers={mockGetMembers}
        houses={[]}
        memberDisplay={{ white: false }}
        removeMembersDisplay={mockRemoveMembersDisplay}
        addMembersDisplay={mockAddMembersDisplay} />
    );
    wrapper.instance().handleOnClick({ name: 'white', members: [] });
    expect(mockAddMembersDisplay).toHaveBeenCalled();
  });
  
  it('should be able to get members data', () => {
    wrapper = shallow(
      <CardContainer
        getHouses={mockGetHouses}
        getMembers={mockGetMembers}
        houses={[]}
        memberDisplay={{ black: true }}
        removeMembersDisplay={mockRemoveMembersDisplay}
        addMembersDisplay={mockAddMembersDisplay} />
    );
    wrapper.instance().handleOnClick({ name: 'white', members: [] });
    expect(mockGetMembers).toHaveBeenCalled();
  });

  describe('mapStateToProps', () => {
    it('should return an object of state data', () => {
      const mockState = {houses: [], memberDisplay: {}};
      expect(mapStateToProps(mockState)).toEqual(mockState);
    });
  });

  describe('mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    it('should call dispatch with getHouses action creator', () => {
      mapped.getHouses();
      expect(mockDispatch).toHaveBeenCalledWith(actions.getHouses());
    });
    
    it('should call dispatch with getMembers action creator', () => {
      mapped.getMembers([]);
      expect(mockDispatch).toHaveBeenCalledWith(actions.getMembers([]));
    });
    
    it('should call dispatch with removeMembersDisplay action creator', () => {
      mapped.removeMembersDisplay('white');
      expect(mockDispatch)
        .toHaveBeenCalledWith(actions.removeMembersDisplay('white'));
    });

    it('should call dispatch with addMembersDisplay action creator', () => {
      mapped.addMembersDisplay('white');
      expect(mockDispatch)
        .toHaveBeenCalledWith(actions.addMembersDisplay('white'));
    });
  });
});