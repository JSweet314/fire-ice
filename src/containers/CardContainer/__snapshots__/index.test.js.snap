// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    addMembersDisplay={[Function]}
    getHouses={[Function]}
    getMembers={[Function]}
    houses={
        Array [
            Object {
              "ancestralWeapons": "Lady Forlorn",
              "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
              "founded": "",
              "name": "House Corbray of Heart's Home",
              "seats": "Heart's Home",
              "swornMembers": Array [
                "255",
                "256",
                "447",
                "654",
                "677",
                "681",
                "871",
                "1182",
              ],
              "titles": Array [
                "Lord of Heart's Home",
                "King of the Fingers (historical)",
              ],
              "words": "",
            },
          ]
    }
    memberDisplay={
        Object {
            "white": true,
          }
    }
    removeMembersDisplay={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          ancestralWeapons="Lady Forlorn"
          coatOfArms="Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)"
          display="none"
          founded=""
          handleOnClick={[Function]}
          name="House Corbray of Heart's Home"
          seats="Heart's Home"
          swornMembers={
                    Array [
                              "255",
                              "256",
                              "447",
                              "654",
                              "677",
                              "681",
                              "871",
                              "1182",
                            ]
          }
          titles={
                    Array [
                              "Lord of Heart's Home",
                              "King of the Fingers (historical)",
                            ]
          }
          words=""
/>,
      ],
      "className": "Container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "ancestralWeapons": "Lady Forlorn",
          "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
          "display": "none",
          "founded": "",
          "handleOnClick": [Function],
          "name": "House Corbray of Heart's Home",
          "seats": "Heart's Home",
          "swornMembers": Array [
            "255",
            "256",
            "447",
            "654",
            "677",
            "681",
            "871",
            "1182",
          ],
          "titles": Array [
            "Lord of Heart's Home",
            "King of the Fingers (historical)",
          ],
          "words": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            ancestralWeapons="Lady Forlorn"
            coatOfArms="Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)"
            display="none"
            founded=""
            handleOnClick={[Function]}
            name="House Corbray of Heart's Home"
            seats="Heart's Home"
            swornMembers={
                        Array [
                                    "255",
                                    "256",
                                    "447",
                                    "654",
                                    "677",
                                    "681",
                                    "871",
                                    "1182",
                                  ]
            }
            titles={
                        Array [
                                    "Lord of Heart's Home",
                                    "King of the Fingers (historical)",
                                  ]
            }
            words=""
/>,
        ],
        "className": "Container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "ancestralWeapons": "Lady Forlorn",
            "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
            "display": "none",
            "founded": "",
            "handleOnClick": [Function],
            "name": "House Corbray of Heart's Home",
            "seats": "Heart's Home",
            "swornMembers": Array [
              "255",
              "256",
              "447",
              "654",
              "677",
              "681",
              "871",
              "1182",
            ],
            "titles": Array [
              "Lord of Heart's Home",
              "King of the Fingers (historical)",
            ],
            "words": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
